### ISSUE TITLE
Why GroupNorm and BatchNorm have weights ans bias?

### ISSUE THREAD

**Issue by MR-hyj** [pytorch/pytorch#75145]
To my best knowledge, both GroupNorm and BatchNorm perform similar functions that leverage the input $x$ to normal distribution, like this:
```python
var = x.var()
x = x - x.mean()
x = x/var
```

So why does `torch.nn.GroupNorm` have `self.weight` and `self.bias` ?
I noticed this problem when I debugging a binary network in which the weight ans bias of GroupNorm is always zero.
```python
class BiGroupNorm(torch.nn.GroupNorm):
    """
    this is a binary implementation of torch.nn.GroupNorm
    """
    def __init__(self, num_groups: int, num_channels: int, eps: float = 1e-5, affine: bool = True, device=None,
                 dtype=None) -> None:
        super(BiGroupNorm, self).__init__(num_groups, num_channels, eps, affine, device, dtype)
        self._logger = logging.getLogger()

    def forward(self, input_: torch.Tensor) -> torch.Tensor:

        self.weight = Parameter(self.weight - self.weight.mean())
        self.weight = Parameter(BinaryQuantize().apply(self.weight))
        self.bias = Parameter(BinaryQuantize().apply(self.bias))
       
        # NOTICE
        # the following two lines suggest weight and bias is always zeros
        self._logger.debug('weight shape of {}: {}, sum of abs weight: {}'.format(self._get_name(), self.weight.shape,
                                                                              torch.sum(torch.abs(self.weight))))
        self._logger.debug('bias shape of {}: {}, sum of abs bias: {}'.format(self._get_name(), self.bias.shape,
                                                                              torch.sum(torch.abs(self.bias))))

        return F.group_norm(
            input_, self.num_groups, self.weight, self.bias, self.eps)

class BinaryQuantize(Function):
    @staticmethod
    def forward(ctx, input_):
        """
        input_[ input_ >0 ] = 1
        input_[ input_ <0 ] = -1
        """
        ctx.save_for_backward(input_)
        out = torch.sign(input_)
        return out

    @staticmethod
    def backward(ctx, grad_output):
        input_ = ctx.saved_tensors
        grad_input = grad_output
        grad_input[input_[0].gt(1)] = 0 
        grad_input[input_[0].lt(-1)] = 0 
        return grad_input
```

>2022-03-31 16:13:00,360 [DEBUG] root - weight shape of BiGroupNorm: torch.Size([1024]), sum of abs weight: 0.0
2022-03-31 16:13:00,361 [DEBUG] root - bias shape of BiGroupNorm: torch.Size([1024]), sum of abs bias: 0.0 
2022-03-31 16:13:00,931 [DEBUG] root - weight shape of BiGroupNorm: torch.Size([512]), sum of abs weight: 0.0
2022-03-31 16:13:00,932 [DEBUG] root - bias shape of BiGroupNorm: torch.Size([512]), sum of abs bias: 0.0
2022-03-31 16:13:00,933 [DEBUG] root - weight shape of BiGroupNorm: torch.Size([256]), sum of abs weight: 0.0
2022-03-31 16:13:00,933 [DEBUG] root - bias shape of BiGroupNorm: torch.Size([256]), sum of abs bias: 0.0
2022-03-31 16:13:00,342 [DEBUG] root - weight shape of BiGroupNorm: torch.Size([64]), sum of abs weight: 0.0
2022-03-31 16:13:00,344 [DEBUG] root - bias shape of BiGroupNorm: torch.Size([64]), sum of abs bias: 0.0

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by vadimkantorov** [comment]
BatchNorm / GroupNorm are typically fused with an affine transform, so `weight` / `bias` are for this post-normalization affine transform. Affine transform can be disabled IIRC by setting `affine=False` in constructor args

**Comment by albanD** [comment]
Hi,

We use github issues only for bugs or feature requests.
Please use the forum to ask questions: https://discuss.pytorch.org/

The answer from Vadim above is accurate for your question.

**Comment by MR-hyj** [comment]
> Hi,
> 
> We use github issues only for bugs or feature requests. Please use the forum to ask questions: https://discuss.pytorch.org/
> 
> The answer from Vadim above is accurate for your question.

I'm sorry for causing you any trouble. I'm deleting this issue. :)
